airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -144,30 +144,39 @@ public void decrement(final long stateId, final long count) {
   public List<PartialAirbyteMessage> flushStates() {
     final List<PartialAirbyteMessage> output = new ArrayList<>();
     Long bytesFlushed = 0L;
-    for (final Map.Entry<StreamDescriptor, LinkedList<Long>> entry : streamToStateIdQ.entrySet()) {
-      // Remove all states with 0 counters.
-      // Per-stream synchronized is required to make sure the state (at the head of the queue)
-      // logic is applied to is the state actually removed.
-      synchronized (this) {
+    synchronized (this) {
```

Comment:
note: I'm pretty sure iterating over a concurrent hash map is thread safe as the iterator returns a snapshot of the map when it's first created.

the synchronized block was originally scoped to the `stateIdQ` since this is a block of operations we need to do as a transaction across a bunch of queues.

this code is fine as is. just sharing some background context.